### DISCUSSION TITLE
Type checking of getId params

### DISCUSSION THREAD

**Discussion by nihilenz** [react-navigation/react-navigation#10515] [Q&A] [answered]
Hello,
when using https://reactnavigation.org/docs/screen/#getid, is it possible to type the `params` object in TypeScript?

I defined a `type` of screen param list as suggested in https://reactnavigation.org/docs/typescript/ (and also defined an `interface` for each screen params object) to use it to annotate `useNavigation` or get props via `NativeStackScreenProps`. 

But how can I do type checking with the `params` object of `getId`?

Thanks

**Comment by satya164** [comment, marked as answer]
If you have implemented type-checking like mentioned in the docs, there's nothing special to type check `getId`. It will already have proper types.

**Reply by nihilenz** [reply]
Sorry, my mistake, I forgot to annotate `createNativeStackNavigator`
